ppx/tools/ppx_deriving_tools.ml

```diff
@@ -732,52 +586,6 @@ module Conv = struct
                :: next)
          in
          pexp_match ~loc x cases
-
-       method! derive_of_type_declaration td =
```

Comment:
what exactly are we deleting? aren't we reducing the scope of what's supported?

Reply:
I’m going to work a bit more in the last commit but the gist is — instead of generating a separate function for polyvariants which decodes into None on unmatched tag, we extend error type to have a case for unmatched tag. This is why we removed code here.